lib/TreeSearch.js

```diff
@@ -5,50 +5,55 @@ const TreeSearch = {};
 // Returns a function that performs a preorder search on the tree for the given
 // simplification function
 TreeSearch.preOrder = function(simplificationFunction) {
-  return function (node) {
-    return search(simplificationFunction, node, true);
+  return function (node, scope={}) {
+    return search(simplificationFunction, node, true, scope);
   };
 };
 
 // Returns a function that performs a postorder search on the tree for the given
 // simplification function
 TreeSearch.postOrder = function(simplificationFunction) {
-  return function (node) {
-    return search(simplificationFunction, node, false);
+  return function (node, scope={}) {
+    return search(simplificationFunction, node, false, scope);
   };
 };
 
 // A helper function for performing a tree search with a function
-function search(simplificationFunction, node, preOrder) {
+function search(simplificationFunction, node, preOrder, scope={}) {
   let status;
 
   if (preOrder) {
-    status = simplificationFunction(node);
+    status = simplificationFunction(node, scope);
     if (status.hasChanged()) {
       return status;
     }
   }
 
-  if (Node.Type.isConstant(node) || Node.Type.isSymbol(node)) {
+  if (Node.Type.isConstant(node)) {
     return Node.Status.noChange(node);
   }
+  // Moved isUnaryMinus check above Symbol check to address symbol nested inside unaryminus
```

Comment:
try to avoid saying things like "moved" since future people looking at this code will only know the current state, not the past and that'd be confusing!

you could instead say something like "isUnaryMinus needs to be checked before Symbol because ..."

Though I'm confused why it has to be first. What test would fail if you checked symbol first? A unary minus is not a symbol, and unary minus recurses on its child! (let me know if I'm being confusing, not sure if I'm explaining this well ^^)